airbyte-cdk/python/connector_builder/connector_builder_handler.py

```diff
@@ -0,0 +1,42 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from datetime import datetime
+from typing import Any, Mapping, Union
+
+from airbyte_cdk.models import AirbyteMessage, AirbyteRecordMessage
+from airbyte_cdk.sources.declarative.manifest_declarative_source import ManifestDeclarativeSource
+from airbyte_cdk.utils.traced_exception import AirbyteTracedException
+
+
+class ConnectorBuilderHandler:
+    def __init__(self, source: ManifestDeclarativeSource):
+        self.source = source
+
+    def list_streams(self) -> AirbyteRecordMessage:
+        raise NotImplementedError
+
+    def stream_read(self, command_config) -> AirbyteRecordMessage:
+        raise NotImplementedError
+
+    @staticmethod
+    def _emitted_at():
+        return int(datetime.now().timestamp()) * 1000
+
+    def resolve_manifest(self) -> Union[AirbyteMessage, AirbyteRecordMessage]:
+        try:
+            return AirbyteRecordMessage(
+                data={"manifest": self.source.resolved_manifest},
+                emitted_at=self._emitted_at(),
+                stream="",
+            )
+        except Exception as exc:
+            error = AirbyteTracedException.from_exception(exc, message="Error resolving manifest.")
+            return error.as_airbyte_message()
+
+    def handle_request(self, config: Mapping[str, Any]) -> Union[AirbyteMessage, AirbyteRecordMessage]:
+        command = config.get("__command")
```

Comment:
nit: I'm wondering if this doesn't leak a little bit of the API in the handler layer. For example, if we say we want to have a connector builder as an HTTP service, would this method be used or we would have another mechanism to call the proper method? My guess is that we would have different endpoints like `/manifest/resolve` and the API layer would call `handler. resolve_manifest(...)` directly. Based on that, `handle_request` is probably part of our API and not our handle and this could be moved to `airbyte-cdk/python/source_declarative_manifest/main.py`

Reply:
I buy it. Updated.